src/misc/cross.cpp

```diff
@@ -439,6 +440,224 @@ FILE *fopen_wrap_ro_fallback(const std::string &filename, bool &is_readonly)
 	return fp;
 }
 
+bool wild_match(const char *haystack, char *needle)
```

Comment:
Should be great for a unit test.

Reply:
Added unit tests for this function. Done.